### DISCUSSION TITLE
could nocodb consider use dual licence?

### DISCUSSION THREAD

**Discussion by joebnb** [nocodb/nocodb#381] [Discord Threads]
gnu contains some risk in commercial could this to be supported

**Comment by zpl** [comment]
There is also problem for me. I can't use AGPL projects for my work. but probably I can buy commercial license

**Comment by o1lab** [comment]
AGPL doesnt stop you from commercial usage. Only if you change nocodb source code - you have to open soruce the code. And if included within another project that source code becomes AGPL.

**Reply by kawpii** [reply]
I am not lawyer but please read the [license](https://github.com/nocodb/nocodb/blob/master/LICENSE)

> **which include larger works using a licensed work, under the same license.** 
>  **Network use is distribution**

As I think any projects (source code) that communicate with this project through network(http) should also disclose their source, thus eliminating use with proprietary code, so I suppose license like [LGPL](https://choosealicense.com/licenses/lgpl-3.0/) may be better appropriate.

**Reply by joebnb** [reply]
i cant use gnu licece code,beacuse if make change it have to open source, it's a hard work from some custom develop

**Reply by ghost** [reply]
GPL - If you link your product with a GPL library, you can use the product internally but not ship it to customers.  If you do, you have to publish your product code too.

AGPL - Same as above except that you can't even make your product available to your customers as a SaaS service.  If you do, you have to publish your product code too.

Hence this question:  Is there a commercial license for NocoDB?
